pkg/controller/operators/olm/apiservices.go

```diff
@@ -138,8 +247,8 @@ func (a *Operator) installAPIServiceRequirements(desc v1alpha1.APIServiceDescrip
 			Selector: depSpec.Selector.MatchLabels,
 		},
 	}
-	// TODO(Nick): ensure service name is a valid DNS name
-	// replace all '.'s with "-"s to convert to a DNS-1035 label
+
+	// Replace all '.'s with "-"s to convert to a DNS-1035 label
 	service.SetName(strings.Replace(apiServiceName, ".", "-", -1))
```

Comment:
since we use this in multiple places now, we should probably have a helper `apiServiceNameToServiceName`